driver-core/src/main/com/mongodb/internal/connection/DefaultServer.java

```diff
@@ -67,41 +85,67 @@
         serverListener.serverOpening(new ServerOpeningEvent(this.serverId));
 
         this.serverMonitor = serverMonitor;
+        operationCount = trackOperationCount ? new AtomicInteger() : null;
```

Comment:
Even the slight complication of the trackOperationCount parameter seems unnecessary since in production code the value is always true.  I suggest that the parameter be removed.

Reply:
I introduced the parameter because some tests cast `server.getConnection()` to `TestConnection`. Any other approach to solve this seems to be even more ugly, but I may simply be failing see a better approach. Do you see a less invasive and/or more elegant approach?

Reply:
I'll think about it. Can you link me to an example?

Reply:
Example: https://github.com/stIncMale/mongo-java-driver/blob/JAVA-3926/driver-core/src/test/unit/com/mongodb/internal/connection/DefaultServerSpecification.groovy#L415

Reply:
Only idea I have is to make `OperationCountTrackingConnection` VisibleForTesting and add a `getWrapped` method to it.  Then callers can cast to `OperationCountTrackingConnection`.  Not sure that's better though.  What do you think?

Reply:
That is an approach I thought of, and I prefer the constructor argument more: it is cleaner and results in smaller code changes.

Reply:
ok